dashboard/src/modules/components/OverviewComponent/SavedRunsComponent.jsx

```diff
@@ -0,0 +1,186 @@
+import "./index.less";
+
+import {
+  ActionsColumn,
+  InnerScrollContainer,
+  OuterScrollContainer,
+  TableComposable,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from "@patternfly/react-table";
+import { findNoOfDays, formatDateTime } from "utils/dateFunctions";
+import {
+  publishDataset,
+  setSelectedRuns,
+  updateDataset,
+} from "actions/overviewActions";
+import { useDispatch, useSelector } from "react-redux";
+
+import { ProgressBar } from "./common-component";
+import React from "react";
+
+const SavedRunsComponent = () => {
+  const dispatch = useDispatch();
+  const { savedRuns, selectedRuns } = useSelector((state) => state.overview);
+  const loginDetails = useSelector((state) => state.userAuth.loginDetails);
+
+  /* Selecting */
+  const areAllRunsSelected =
+    savedRuns?.length > 0 && savedRuns?.length === selectedRuns?.length;
+  const selectAllRuns = (isSelecting) => {
+    dispatch(setSelectedRuns(isSelecting ? savedRuns.map((r) => r) : []));
+  };
+  const onSelectRuns = (run, __rowIndex, isSelecting) => {
+    const otherSelectedRuns = selectedRuns.filter(
+      (r) => r.resource_id !== run.resource_id
+    );
+    const c = isSelecting ? [...otherSelectedRuns, run] : otherSelectedRuns;
+    dispatch(setSelectedRuns(c));
+  };
```

Comment:
I know that I've suggested using a `Set` instead of any `Array` for keeping the runs list since each entry is supposed to be unique, but it occurs to me that you might want to try a `Map` instead (in a `Map`, the keys are all unique):  here you could use `.set()` to add the value (without having to check if it is already there), and you could use `.delete()` to remove it (without having to use `.filter()`).

And, for `isRowSelected()`, below, you could use `.has()` instead of `.filter()` and `.length`.